legacy-sdk/whirlpool/tests/utils/litesvm.ts

```diff
@@ -0,0 +1,1230 @@
+  const newClock = new Clock(
+    newSlot,
+    currentClock.epochStartTimestamp,
+    currentClock.epoch,
+    currentClock.leaderScheduleEpoch,
+    newTimestamp,
+  );
+  litesvm.setClock(newClock);
+  // Expire the current blockhash to force generation of a new one
+  // This ensures subsequent transactions use a fresh blockhash
+  litesvm.expireBlockhash();
+  console.log(
+    `⏰ Warped clock: +${seconds}s (slot: ${currentClock.slot} -> ${newSlot}, time: ${currentClock.unixTimestamp} -> ${newTimestamp})`,
+  );
+}
+
+/**
+ * Advance to the next epoch in the blockchain
+ * This is needed for tests that depend on epoch changes (e.g., transfer fee updates)
+ */
+export function advanceEpoch(): void {
+  const litesvm = getLiteSVM();
+  const currentClock = litesvm.getClock();
+  // Advance to next epoch
+  const newEpoch = currentClock.epoch + 1n;
+  // Advance slot significantly (typical epoch has ~432000 slots)
+  const slotsToAdvance = 432000n;
+  const newSlot = currentClock.slot + slotsToAdvance;
+  // Advance timestamp proportionally (assuming ~400ms per slot)
+  const secondsToAdvance = 432000 * 0.4; // ~172800 seconds (2 days)
+  const newTimestamp =
+    currentClock.unixTimestamp + BigInt(Math.floor(secondsToAdvance));
+  const newClock = new Clock(
+    newSlot,
+    newTimestamp, // new epoch start timestamp
+    newEpoch,
+    newEpoch, // leaderScheduleEpoch typically matches epoch
+    newTimestamp,
+  );
+  litesvm.setClock(newClock);
+  litesvm.expireBlockhash();
+  console.log(
+    `🔄 Advanced epoch: ${currentClock.epoch} -> ${newEpoch} (slot: ${currentClock.slot} -> ${newSlot})`,
+  );
+}
+
+/**
+ * Poll for a condition to be met, with retries and account reload for state synchronization
+ * This is useful for LiteSVM which sometimes has delayed state updates
+ *
+ * @param checkFn - Async function that returns the result to check
+ * @param conditionFn - Function that checks if the condition is met on the result
+ * @param accountToReload - Optional account address to force reload before each check
+ * @param connection - Connection to use for account reload
+ * @param maxRetries - Maximum number of retries (default: 10)
+ * @param delayMs - Delay between retries in milliseconds (default: 10)
+ * @returns The result once the condition is met
+ * @throws Error if condition is not met after max retries
+ */
+export async function pollForCondition<T>(
+  checkFn: () => Promise<T>,
+  conditionFn: (result: T) => boolean,
+  options?: {
+    accountToReload?: PublicKey;
+    connection?: anchor.web3.Connection;
+    maxRetries?: number;
+    delayMs?: number;
+  },
+): Promise<T> {
+  const maxRetries = options?.maxRetries ?? 10;
+  const delayMs = options?.delayMs ?? 10;
+  let result: T;
+  for (let retry = 0; retry < maxRetries; retry++) {
+    // Force account reload if specified
+    if (options?.accountToReload && options?.connection) {
+      await options.connection.getAccountInfo(options.accountToReload);
```

Comment:
curious, does this refresh the account in memory under the hood or smth? so we just need to call it and not necessarily use the resulting data?